kong/plugins/consumer-rate-limiting/timeutils.lua

```diff
@@ -10,11 +10,15 @@ function timeutils.get_timestamp_keys(timestamp)
   }
 end
 
-function timeutils.PeriodGenerator()
+function timeutils.PeriodGenerator(cadence)
   local self = {}
 
   function self.get_period(timestamp)
-    return timeutils.get_timestamp_keys(timestamp).months
+    if timeutils.get_timestamp_keys(timestamp)[cadence] ~= nill then
+      return timeutils.get_timestamp_keys(timestamp)[cadence]
```

Comment:
Could you add checking, if the returned value is not a nil? It could happend, that someone defines an not existing cadence and nil will be returned.